src/app.js

```diff
@@ -59,7 +59,11 @@ function listen (koaApp, config, ledgers) {
   // Start a coroutine that connects to the backend and
   // subscribes to all the ledgers in the background
   co(function * () {
-    yield backend.connect()
+    const success = yield backend.connect()
+    if (!success) {
+      log('app').error('backend initialization has failed - the connector will now exit')
+      process.exit(1)
```

Comment:
The main difference I see between throwing an exception and returning a value is that with an exception, if the calling method forgets to catch the exception, it will bubble up to the top of the call stack, but if it forgets to check the response, nothing will happen. But as long as the app checks the response from `connect`, I don't see an issue with this.